lib/queuePopulator/LogReader.js

```diff
@@ -407,25 +416,29 @@ class LogReader {
             logStats.nbLogRecordsRead += 1;
 
             this._setEntryBatch(entriesToPublish);
-            record.entries.forEach(entry => {
+            /**
+             * waiting for all sync and async extentions
+             * to finish processing before emptying the batch
+             */
+            async.each(record.entries, async entry => {
```

Comment:
It may be important to process the entries in series (as is, only call the next async `_processLogEntry` when the previous one completes), in which case using `async.eachSeries` is needed. Otherwise, it could still be nicer to use `async.eachLimit` to limit the parallelism.